digital_land/utils/add_data_utils.py

```diff
@@ -48,3 +51,135 @@ def clear_log(collection_dir, endpoint):
+            os.path.join(converted_dir, endpoint_resource_info["resource"] + ".csv")
+        )
+    except FileNotFoundError:
+        converted_resource_df = pd.read_csv(endpoint_resource_info["resource_path"])
+
+    # Find columns that are in resource that aren't in the column field log
+    unmapped_columns = [
+        column
+        for column in converted_resource_df.columns
+        if column not in column_field_df["column"].values
+    ]
+
+    if len(unmapped_columns) > 0:
+        column_field_summary += "\n"
+        column_field_summary += ", ".join(unmapped_columns)
+    else:
+        column_field_summary += "\nNo unmapped columns!"
+
+    column_field_summary += "\n\nUnmapped Fields:"
+    mapped_fields = column_field_df["field"].values
+    specification = Specification(specification_dir)
+    dataset_fields = specification.dataset_field[pipeline]
+
+    # remove fields from dataset fields that are autogenerated by pipeline
+    exclude_fields = ["entity", "organisation", "prefix"]
+    if pipeline != "tree":
+        exclude_fields.append("point")
+    dataset_fields = [field for field in dataset_fields if field not in exclude_fields]
+
+    unmapped_fields = [field for field in dataset_fields if field not in mapped_fields]
+    if len(unmapped_fields) > 0:
+        column_field_summary += "\n"
+        column_field_summary += ", ".join(unmapped_fields)
+    else:
+        column_field_summary += "\nNo unmapped fields!"
+
+    if "reference" not in mapped_fields:
+        raise ValueError(
+            "Reference not found in the mapped fields - does this need mapping?"
+        )
+
+    return column_field_summary
+
+
+def get_issue_summary(endpoint_resource_info, issue_dir):
+    issue_summary = ""
+    issue_summary += (
+        "\n======================================================================"
+    )
+    issue_summary += "\nIssue Summary"
+    issue_summary += (
+        "\n======================================================================"
+    )
+
+    issue_df = pd.read_csv(
+        os.path.join(issue_dir, endpoint_resource_info["resource"] + ".csv")
+    )
+
+    issue_summary += "\n"
+    issue_summary += issue_df.groupby(["issue-type", "field"]).size().to_string()
+
+    return issue_summary
+
+
+def get_entity_summary(endpoint_resource_info, output_path, issue_dir):
+    entity_summary = ""
+    entity_summary += (
+        "\n======================================================================"
+    )
+    entity_summary += "\nEntity Summary"
+    entity_summary += (
+        "\n======================================================================"
+    )
+
+    # Get number of entities in resource from transformed file
+    transformed_df = pd.read_csv(output_path)
+    entity_count = transformed_df["entity"].nunique()
+    entity_summary += f"\nNumber of existing entities in resource: {entity_count}"
+
+    # Get new entities by looking at unknown entity issues
```

Comment:
I could be wrong but does it still have access to the issue log itself? could this be pulled out of there?

Reply:
the issue log is created inside the `pipeline_run()` function, we just pass in the issue dir, so we don't have access to the object outside of the function